vsphere/resource_vsphere_compute_cluster.go

```diff
@@ -1455,6 +1505,60 @@ func expandVsanDatastoreConfig(d *schema.ResourceData, meta interface{}) (*vsant
 	return conf, nil
 }
 
+func buildVsanStretchedClusterReq(d *schema.ResourceData, cluster types.ManagedObjectReference) (*vsantypes.VSANVcConvertToStretchedCluster, error) {
+	log.Printf("[DEBUG] building vsan stretched cluster request...")
+	conf := d.Get("vsan_stretched_cluster_conf").(*schema.Set).List()[0].(map[string]interface{})
+
+	if conf["preferred_fault_domain_host_ids"] == "" {
+		return nil, fmt.Errorf("preferred_fault_domain_host_ids field is required in vsan_stretched_cluster_conf when enabling stretched cluster!")
+	}
+
+	if conf["secondary_fault_domain_host_ids"] == "" {
+		return nil, fmt.Errorf("secondary_fault_domain_host_ids field is required in vsan_stretched_cluster_conf when enabling stretched cluster!")
+	}
+
+	if conf["witness_node"] == "" {
+		return nil, fmt.Errorf("witness_node field is required in vsan_stretched_cluster_conf when enabling stretched cluster!")
+	}
+
+	witness_node := []string{}
+	witness_node = append(witness_node, conf["witness_node"].(string))
+	witness := structure.SliceStringsToManagedObjectReferences(witness_node, "HostSystem")
```

Comment:
This can be a oneliner:
```suggestion
	witness := structure.SliceStringsToManagedObjectReferences([]string{conf["witness_node"].(string)}, "HostSystem")
```